sklearn/manifold/_t_sne.py

```diff
@@ -714,16 +733,19 @@ def _fit(self, X, skip_num_points=0):
                 if self.verbose:
                     print("[t-SNE] Computing pairwise distances...")
 
-                if self.metric == "euclidean":
-                    distances = pairwise_distances(X, metric=self.metric,
-                                                   squared=True)
-                else:
-                    distances = pairwise_distances(X, metric=self.metric,
-                                                   n_jobs=self.n_jobs)
+                distances = pairwise_distances(X, metric=self.metric,
```

Comment:
Why the switch here?

Reply:
By perform squaring [a little further down](https://github.com/scikit-learn/scikit-learn/pull/17662/files/88340e6786e1a8be09a2bb8ea9c22eeadd4dd6ac#diff-e9ec8beb584b0141adb0b124e88fbd40R744) instead, it allowed me to combine the legacy case with the general case, which I felt was a bit cleaner.

Reply:
pairwise_distances with euclidean_distances is know to be slower with n_jobs > 1 so we should keep the branching.

Also it's better to pass `squared` for euclidean since it first computes the squared distance matrix. Otherwise you'll compute the squared distance matrix, then take the square root and then square it again.

Reply:
Thank you for the insight. I'll add a comment to make this reasoning clearer. :)